### DISCUSSION TITLE
a way to let the pixel displays on the spike/invetor hubs work in a screen buffer way

### DISCUSSION THREAD

**Discussion by MimiVRC** [pybricks/support#1764] [Ideas]
![image](https://github.com/user-attachments/assets/064e1da3-cf06-4bbc-b152-47840b00a4e8)
this would be the best way that I know of to use the pixel displays. The problem with how it works now is it does each function shown and the LEDs are not fast enough to not flicker when doing it this the shown method

to avoid having them flash or needing to address every LED in every elseif, it would be nice if we had a new node that lets you write into a "buffer" variable what the LED matrix screen should display and when you want it to display you use a "buffer display" node with which buffer you want to display. this lets you every frame clear the screen buffer, as shown in my image, and then set up the buffer, then display it. the screen only changes the LEDs that actually change

other uses of this idea: you create a symbol on start that you save to a buffer, and then at any time you can display it from the X and Y for the top left of the buffer and coordinates on the LED matrix, then move it around by changing the display coordinates

**Comment by laurensvalk** [comment]
Thank you for the suggestion. We have something like this in Python, but there isn't an icon editor in the block language yet.

> The problem with how it works now is it does each function shown and the LEDs are not fast enough to not flicker when doing it this the shown method

I think speed isn't necessarily the problem. Rather, it might be updating _too fast_. Your code is updating the display thousands of times per second, which (perhaps unintuitively) makes the (very small) time between command much more visible.

I would suggest adding a small wait at the end of the loop, such as `25 ms`. That makes it update at 40 fps, which is more than enough for the human eye. And it should flicker less, since the display will be idle most of the time.
